### ISSUE TITLE
Building mpfr breaks on x64-windows

### ISSUE THREAD

**Issue by justinjk007** [Microsoft/vcpkg#2448]
![image](https://user-images.githubusercontent.com/12916795/34361182-e48e138e-ea35-11e7-9987-27e71fa96ad9.png)

![image](https://user-images.githubusercontent.com/12916795/34361184-eb62056c-ea35-11e7-8b06-6d081c57d0c3.png)

I'm not sure where to find any other log files if they exist

**Comment by jasjuang** [comment]
It is because mpfr 4.0.0 is released. I will do a PR shortly to update and fix your problem.

**Comment by justinjk007** [comment]
thank you very much

**Comment by jasjuang** [comment]
I got an unresolved external symbol error when building mpfr 4.0.0. According to the website now mpfr depends on a newer version of gmp, but the gmp equivalent on windows mpir is still not updated. I am not sure how to solve this. Maybe @xoviat can help.

@justinjk007 The error message said download failed, hence the first thing I do is go to my browser to check what's going on, then I realized mpfr 4.0.0 is released. For some reason mpfr doesn't keep older versions on their website.

**Comment by xoviat** [comment]
Note: a generic C target is available for MPIR here: https://github.com/chfast/mpir/blob/47910ac631c599774577b155a33fa3a372155c1d/CMakeLists.txt

If this issue is important to you, please down vote this post: https://github.com/wbhart/mpir/pull/215#issuecomment-353179717

It may change minds.

**Comment by jasjuang** [comment]
@xoviat Even if they refuse to support cmake, doesn't at least the cmake patch in vcpkg kind of work around the problem? I think we need to try building mpir with `vcpkg_from_github` (doesn't look trivial to me at the moment) and use the `--head` flag and see whether it will make mpfr 4.0.0 build. If it builds then we can request for a new release to mpir.

**Comment by xoviat** [comment]
No. The correct way to build MPIR is to use the MSBUILD files in the repository. What's practically possible is to build directly from the fork I linked.

**Comment by jasjuang** [comment]
@xoviat I looked more carefully in the mpir portfile and you are right, it's currently build with msbuild, but still it's worth a try to see whether the `--head` version of mpir actually makes mpfr 4.0.0 to build. 

Another thing is for some reason the git release tags on the mpir repo is only up to 2.7.0 instead of 3.0.0, so before we make the transition to `vcpkg_fron_github` this has to be solved first. I already filed an issue and hopefully they will respond soon.
